src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,96 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+import { getPostDetails,
+  getPostComments,
+  removeCommentById,
+  createComment } from '../../api/posts';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+export const PostDetails = ({ postId }) => {
+  const [post, setPost] = useState(null);
+  const [comments, setComments] = useState([]);
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+  const loadPosts = async() => {
+    setPost(await getPostDetails(postId));
+  };
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  const loadComments = async() => {
+    setComments(await getPostComments(postId));
+  };
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
+  const removeComment = async(id) => {
+    await removeCommentById(id);
+    loadComments();
+  };
+
+  const addComment = async({ name, email, body }) => {
+    // eslint-disable-next-line
+    await createComment({postId, name, email, body});
+    loadComments();
+  };
+
+  useEffect(() => {
+    loadPosts();
+    loadComments();
+  }, [postId]);
+
+  const handleHideComments = (event) => {
+    const button = event.target;
+    const commentsList = event.target.nextElementSibling;
 
-        <li className="PostDetails__list-item">
+    commentsList.hidden = !commentsList.hidden;
+    button.innerText = 'Show comments';
+  };
```

Comment:
Let's use conditional rendering for this purpose. Don't use DOM methods while using react